caikit/core/task.py

```diff
@@ -194,3 +238,8 @@ def decorator(cls: Type[TaskBase]) -> Type[TaskBase]:
         return cls
 
     return decorator
+
+
+def _is_iterable_type(type_: Type) -> bool:
+    """Checks if a type is iterable"""
+    return isinstance(type_, collections.abc.Iterable)
```

Comment:
there are a lot of checks in `isa.py` but I can't import that here or it'll cycle